### ISSUE TITLE
Commit 5556265559d4 ("rule: rule-file severity has precedence") triggers false positives

### ISSUE THREAD

**Issue by antgres** [priv-kweihmann/oelint-adv#807]
Heyo,

with the addition of 5556265559d4 ("rule: rule-file severity has precedence") my pipeline fails with the following error:

```
recipes-kernel/imx-firmware/imx-firmware_8.28.bb:46:error:oelint.vars.mispell.unknown:'NO_GENERIC_LICENSE' is unknown. Consider adding it via --constantmods [branch:true]
```

for the snippet

```
...

NO_GENERIC_LICENSE[Firmware-nxp-imx-firmware] = "COPYING"
...
```

I think this is an error because [NO_GENERIC_LICENSE is a known value](https://docs.yoctoproject.org/next/dev-manual/licenses.html#copying-non-standard-licenses).

--Anton

**Comment by priv-kweihmann** [comment]
NO_GENERIC_LICENSE never has been part of the data (oelint-data), so this ain't a regression (also because the latest update is not linked to that at all).

The variable might be documented by core, which doesn't mean it's actually set as by the code (see constant guide for more explanation).

IMO that ain't a big of the linter, so I don't plan on fixing this
